lib/improver/tests/utilities/test_statistical_operations.py

```diff
@@ -0,0 +1,278 @@
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        expected = set_reference_probabilities()
+
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+    def test_values_inverse_ordering(self):
+        """
+        Test that interpolated probabilities at given topography heights are
+        sensible when we use the inverse_ordering set to True. This is for
+        situations in which the values associated with the percentiles increase
+        in the opposite direction, e.g. 0 % = 100m, 20% = 50m, etc.
+
+        In this situation we expect the lowest points to have a probability of
+        1, and the highest points to have probabilities of 0. The probabilities
+        between should be the inverse of what is found in the usual case.
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        # Invert the values associated with the percentiles.
+        percentiles_cube.data = np.flipud(percentiles_cube.data)
+        expected = set_reference_probabilities()
+        expected = 1.0 - expected
+
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+    def test_equal_percentiles(self):
+        """
+        Test for sensible behaviour when some percentile levels are equal
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        percentiles_cube.data[0].fill(300.)
+        expected = set_reference_probabilities()
+        expected[np.where(expected < 0.25)] = 0.
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+    def test_equal_percentiles_inverse_ordering(self):
+        """
+        Test for sensible behaviour when some percentile levels are equal
+        in the case of inverse ordering (as described above).
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        percentiles_cube.data[0].fill(300.)
+        # Invert the values associated with the percentiles.
+        percentiles_cube.data = np.flipud(percentiles_cube.data)
+        expected = set_reference_probabilities()
+        expected[np.where(expected <= 0.25)] = 0.
+        expected = 1.0 - expected
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+
+class Test_process(IrisTest):
```

Comment:
I think you need a few more unit tests here, including ones for:
* Test raises the warning
* Test get the right answer if cubes are in different units
* Test what happens when you have leading dimensions on your cube and make sure these don't get lost. See PR #407 for some kind of example of these unit tests and ways of making sure you promote any coords back to dim coords.

Reply:
* Added warning test to existing 3D test.
* Unit conversion tests added for compatible and incompatible units.
* Added dimension checking as suggested.

Reply:
It might be nice to have one process unit test that tests the data is as expected.